core/common/result.hpp

```diff
@@ -103,6 +125,34 @@ namespace kagome::expected {
                             std::forward<ErrorMatch>(error_func));
     }
 
+    /**
+     * This method returns the value contained in the Result, if it was stored,
+     * or throws a NoValueException exception otherwise.
+     */
+    constexpr V tryGetValue() const {
+      return match(
+          [](const Value<V>& v) { return std::move(v.value); },
+          [](auto &_) -> V { throw NoValueException(); });
+    }
+
+    /**
+     * This method returns the error contained in the Result, if
+     * it was stored, or throws a NoErrorException exception otherwise.
+     */
+    constexpr E tryGetError() const {
+      return match(
+          [](const Error<E>& e) { return std::move(e.error); },
+          [](auto &_) -> E { throw NoErrorException(); });
+    }
+
+    /**
+     * Returns true if the result contains a value, else returns false
+     */
+    constexpr operator bool() const noexcept {
+      return match([](const Value<V> &v) { return true; },
```

Comment:
what if Result does not contain anything?
like
```
Result<int, int> r;
if(r) {
 ... 
}
```

I think we need to get rid of operator bool, and replace with explicit `has_value` and `has_error`

Reply:
I disagree, implicit convertion to bool is quite logical and convenient.